examples_test.go

```diff
@@ -1,29 +1,30 @@
 package oauth_test
 
 import (
-	"fmt"
-	"os"
+  "fmt"
+  "os"
 
-	"github.com/cli/oauth"
+  "github.com/cli/oauth"
 )
 
 // DetectFlow attempts to initiate OAuth Device flow with the server and falls back to OAuth Web
 // application flow if Device flow seems unsupported. This approach isn't strictly needed for
 // github.com, as its Device flow support is globally available, but it enables logging in to
 // self-hosted GitHub instances as well.
 func ExampleFlow_DetectFlow() {
-	flow := &oauth.Flow{
-		Host:         oauth.GitHubHost("https://github.com"),
-		ClientID:     os.Getenv("OAUTH_CLIENT_ID"),
-		ClientSecret: os.Getenv("OAUTH_CLIENT_SECRET"), // only applicable to web app flow
-		CallbackURI:  "http://127.0.0.1/callback",      // only applicable to web app flow
-		Scopes:       []string{"repo", "read:org", "gist"},
-	}
+  host, err := oauth.NewGitHubHost("https://github.com")
```

Comment:
Given that this is exported in [the Go docs](https://pkg.go.dev/github.com/cli/oauth#example-Flow.DetectFlow),  can we handle this error to demonstrate the new intended usage of this function?
```suggestion
  host, err := oauth.NewGitHubHost("https://github.com")
  if err != nil {
    panic(err)
  }
```